pkg/util/interrupt/interrupt.go

```diff
@@ -35,6 +42,10 @@ func Chain(handler *Handler, notify ...func()) *Handler {
 	return New(handler.Signal, append(notify, handler.Close)...)
 }
 
+// New creates a new handler that guarantees all notify functions are run after the critical
+// section exits (or is interrupted by the OS), then invokes the final handler. If no final
+// handler is specified, the default final is `os.Exit(0)`. A handler can only be used for
```

Comment:
I'm fine with os.Exit(2)

On Wed, Apr 6, 2016 at 6:02 AM, Rodolfo Carvalho <notifications@github.com>
wrote:

> In pkg/util/interrupt/interrupt.go
> <https://github.com/kubernetes/kubernetes/pull/23643#discussion_r58680474>
> :
>
> > @@ -35,6 +42,10 @@ func Chain(handler *Handler, notify ...func()) *Handler {
> >  	return New(handler.Signal, append(notify, handler.Close)...)
> >  }
> >
> > +// New creates a new handler that guarantees all notify functions are run after the critical
> > +// section exits (or is interrupted by the OS), then invokes the final handler. If no final
> > +// handler is specified, the default final is `os.Exit(0)`. A handler can only be used for
>
> Perhaps os.Exit(2) would be better, since that's the default exit code
> that the Go runtime uses for the signals we're catching?
> Exiting with 0 tells that the process exited under "normal", okay,
> circumstances, which is not the case here?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23643/files/de6e02d8bb179fe6be34d6b44d84ac8ee4b8938e#r58680474>
>